js/jquery.modalDialog.js

```diff
@@ -1113,6 +1113,52 @@
     // we will be hiding all content within the DOM and scrolling them to top.
     // When removing the host window content from the DOM, make the veil opaque to hide it.
     $.modalDialog.veilClass = "dialog-veil";
+    
+    // Initializer functions. The order determines the priority for creating dialog types.
+    var _typeInitializers = [
+    
+        // creates ajax or iframe dialog from url
+        function (settings) {
+            if (settings.url) {
+                if (settings.content) {
+                    throw new Error("Both url and content cannot be specified.");
+                } else if (settings.ajax) {
+                    return new AjaxDialog(settings);
+                } else {
+                    return new IFrameDialog(settings);
+                }
+            }
+            return null;
+        },
+    
+        // creates node dialog
+        function (settings) {
+            if (settings.content) {
+                var $content = $(settings.content);
+                if ($content.length === 0) {
+                    throw new Error("ModalDialog content not found");
+                }
+    
+                settings.content = $content;
+    
+                var dialog = new ModalDialog(settings);
+    
+                if (!settings.destroyOnClose) {
+                    $content.modalDialogInstance(dialog);
+                }
+    
+                return dialog;
+            }
+    
+            return null;
+        }
+    ];
+
+    // Allows for extending the ModalDialog prototype for other types of dialogs
+    // Add an initialization function to the beginning of the _typeInitializers array to create the plugin type
+    $.modalDialog.registerPlugin = function (f) {
+        f.call(this, ModalDialog, _typeInitializers);
```

Comment:
This is interesting. I think you're exposing _typeInitializers here because it gives a plugin author the ability to insert initializer functions anywhere in the order. I'm wondering if we can expose fewer of the internals and provide enough power to manage the precedence of the initializers. This makes the existing order of precedence of the internal functions an implicit part of the API- changing it could break plugins.

As an alternative, we could have the function passed to registerPlugin() return the typeInitializer function, itself- and have a boolean boolean argument indicating if it should go at the beginning or end of the _typeInitializers array.